### PR TITLE
Q&A: Added some basic CSS styling; added accordion effect; added sorting by REPORTED

### PR DESCRIPTION
**Pull request by FlightfulKiwi**
Per the lengthy title, I've implemented the accordion effect and added sorting by whether or not a Question was reported. The same Reporting functionality should exist for answers but the button does not do anything because the API does not seem to hold a REPORTED property for answers. 
I've also updated the Search Bar sorting such that it only initiates sorting after 3 characters have been typed.
I've made a clear attempt at CSS-ing. It is not terrible-terrible but it is not great. Much more work to come on that front.

Remaining features:
- Modal images (main focus of tomorrow)
- Re-rendering when Q/A added (and when Q Reported?)
- CSS
- [Minor] Display 4 questions on load instead of 2
- Go over BDR for any omitted reqs

(I abstained from assigning to anyone specific as I'd like the first available person to tend to this so I can easily be up and running early. I know I could continue working while awaiting review; this is easiest :-D )

### CODE REVIEW

File: client/src/components/QandA/QuestionsList/QuestionCard.jsx [hunk 1]
```diff
@@ -64,46 +66,61 @@ class QuestionCard extends React.Component {
...
-                  <>
-                    <button onClick={this.onCollapseClick}>Collapse answers</button>
-                    <br/><br/>
-                  </>
-            }
-          </>
-          : null
-        }
+        <div id='QA-question-card-question'>
+          <div className='QA-symbol QA-question-symbol'>Q:&nbsp;&nbsp;</div>
```
Comment (Daniel-Ghaly) [inline client/src/components/QandA/QuestionsList/QuestionCard.jsx]: It seems like QA-question-symbol isn't an actual class in your CSS? I think you just need QA-symbol here?

File: client/src/components/QandA/QuestionsList/QuestionCard/QuestionDetails.jsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ const QuestionDetails = ({ asker, date, helpful, reported, question_id, product_
   const newDate = new Date(date).toLocaleDateString('default', options);
 
   return (
-    <section>
+    <section className='QA-question-details'>
```
Comment (Daniel-Ghaly) [inline client/src/components/QandA/QuestionsList/QuestionCard/QuestionDetails.jsx:11]: good use of section tag!

**Review by Daniel-Ghaly — COMMENTED** [review]
Just remove the extra class name and I'll merge!

File: client/src/components/QandA/QuestionsList/QuestionCard/QuestionDetails.jsx (hunk 2 above)
Comment (FlightfulKiwi) [reply client/src/components/QandA/QuestionsList/QuestionCard/QuestionDetails.jsx:11]: Thanks, I still don't fully comprehend it but didn't wanna only use DIVs

**Review by Daniel-Ghaly — APPROVED** [review]

**Pull request merged**